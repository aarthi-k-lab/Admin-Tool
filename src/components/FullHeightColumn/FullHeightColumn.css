.stretch-column {
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  flex-basis: 0;
  overflow: auto;
}

.stretch-row {
  display: flex;
  flex-direction: row;
  flex-grow: 1;
}
