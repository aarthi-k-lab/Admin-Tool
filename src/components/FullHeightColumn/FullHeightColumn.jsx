import React from 'react';
import PropTypes from 'prop-types';

import './FullHeightColumn.css';

const FullHeightColumn = ({ children, className }) => (
  <div styleName="stretch-column">
    <section className={className} styleName="stretch-row">
      { children }
    </section>
  </div>
);

FullHeightColumn.defaultProps = {
  className: '',
};

FullHeightColumn.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

export default FullHeightColumn;
