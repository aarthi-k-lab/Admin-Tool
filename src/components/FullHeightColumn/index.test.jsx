import React from 'react';
import renderer from 'react-test-renderer';
import FullHeightColumn from '.';

describe('<FullHeightColumn />', () => {
  test('snapshot test', () => {
    const tree = renderer
      .create(<FullHeightColumn className="class1 class2">content</FullHeightColumn>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
