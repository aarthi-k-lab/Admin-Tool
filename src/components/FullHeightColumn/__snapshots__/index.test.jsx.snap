// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullHeightColumn /> snapshot test 1`] = `
<div
  className="components-FullHeightColumn-FullHeightColumn__stretch-column--3jKIT"
>
  <section
    className="class1 class2 components-FullHeightColumn-FullHeightColumn__stretch-row--1GpNn"
  >
    content
  </section>
</div>
`;
